### ISSUE TITLE
Pre-build wheels don't have xformers._flash_attn

### ISSUE THREAD

**Issue by MatthieuBizien** [facebookresearch/xformers#611]
# 🐛 Bug

When I install the wheel from https://pypi.org/project/xformers/0.0.16rc392/ the module xformers._flash_attn does not exist.

This mean importing xformers.ops fails.

```File /usr/local/lib/python3.8/dist-packages/xformers/ops/__init__.py:8
      1 # Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
      2 #
      3 # This source code is licensed under the BSD license found in the
      4 # LICENSE file in the root directory of this source tree.
      6 import torch
----> 8 from .fmha import (
      9     AttentionMask,
     10     AttentionOp,
     11     AttentionOpBase,
     12     AttentionOpDispatch,
     13     LowerTriangularMask,
     14     MemoryEfficientAttentionCutlassFwdFlashBwOp,
     15     MemoryEfficientAttentionCutlassOp,
     16     MemoryEfficientAttentionFlashAttentionOp,
     17     MemoryEfficientAttentionOp,
     18     MemoryEfficientAttentionTritonFwdFlashBwOp,
     19     TritonFlashAttentionOp,
     20     memory_efficient_attention,
     21     memory_efficient_attention_backward,
     22     memory_efficient_attention_forward,
     23     memory_efficient_attention_forward_requires_grad,
     24 )
     25 from .swiglu_op import (
     26     SwiGLU,
     27     SwiGLUEagerOp,
   (...)
     32     swiglu,
     33 )
     34 from .unbind import get_stack_strides, stack_or_none, unbind

File /usr/local/lib/python3.8/dist-packages/xformers/ops/fmha/__init__.py:10
      6 from typing import Any, Optional, Tuple, Type, Union
      8 import torch
---> 10 from . import cutlass, flash, small_k, triton
     11 from .common import (
     12     AttentionBwOpBase,
     13     AttentionFwOpBase,
   (...)
     22     bmk2bmhk,
     23 )
     24 from .dispatch import _dispatch_bw, _dispatch_fw

File /usr/local/lib/python3.8/dist-packages/xformers/ops/fmha/triton.py:16
     13 from ..common import register_operator
     15 if TYPE_CHECKING or _is_triton_available():
---> 16     from ..._flash_attn.flash_attn_triton import (
     17         _flash_attn_backward,
     18         _flash_attn_forward,
     19     )
     21     triton_flash_backward = _flash_attn_backward
     22     triton_flash_forward = _flash_attn_forward

ModuleNotFoundError: No module named 'xformers._flash_attn'
```
## Expected behavior

Wheels should be compiled with Cuda support

## Environment

Please copy and paste the output from the
environment collection script from PyTorch
(or fill out the checklist below manually).

You can run the script with:

```bash
# For security purposes, please check the contents of collect_env.py before running it.
python -m torch.utils.collect_env
```

```
$ python -m torch.utils.collect_env 
Collecting environment information...
PyTorch version: 1.13.0+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 8.4.0-3ubuntu2) 8.4.0
Clang version: Could not collect
CMake version: version 3.25.0
Libc version: glibc-2.31

Python version: 3.8.10 (default, Nov 14 2022, 12:59:47)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.15.0-1022-aws-x86_64-with-glibc2.29
Is CUDA available: True
CUDA runtime version: 11.6.124
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA A10G
Nvidia driver version: 510.73.08
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] ema-pytorch==0.0.10
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.23.4
[pip3] torch==1.13.0
[pip3] torchvision==0.13.1+cu116
[conda] No relevant packages
```

**Comment by danthe3rd** [comment]
Do you have an idea @AbdBarho ?

**Comment by AbdBarho** [comment]
This is really weird:
```sh
$ zipinfo ./xformers-0.0.16rc392-cp310-cp310-manylinux2014_x86_64.whl | grep -c _flash_attn
29

$ zipinfo ./xformers-0.0.16rc392-cp38-cp38-manylinux2014_x86_64.whl | grep -c _flash_attn
0
```

The build process is identical to all python versions.

I will see if I can find it out whats happening here...

**Comment by AbdBarho** [comment]
Side note: `pip install triton` resolves to `1.1.1` on python3.8 and to `2.0.0.dev.....` on python3.10

Also, triton import fails on 3.8:
```py
>>> from xformers.triton.softmax import softmax as triton_softmax
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/_internal/cpython-3.8.15/lib/python3.8/site-packages/xformers/triton/softmax.py", line 14, in <module>
    from xformers.triton.k_softmax import _softmax, _softmax_backward
  File "/opt/_internal/cpython-3.8.15/lib/python3.8/site-packages/xformers/triton/k_softmax.py", line 41, in <module>
    depth: tl.constexpr,
AttributeError: module 'triton.language' has no attribute 'constexpr'
```
